### PR TITLE
Add missing public headers to umbrella header

### PR DESCRIPTION
**Pull request by conradev**
Let me know if I should make any changes!

This fixes Xcode warnings about missing headers when using the framework.

### CODE REVIEW

**Comment by conradev** [comment]
Investigating the build failure…

File: src/async/imap/MCAsyncIMAP.h [hunk 1]
```diff
@@ -12,6 +12,8 @@
 
 #include <MailCore/MCIMAPAsyncSession.h>
 #include <MailCore/MCIMAPOperation.h>
+#include <MailCore/MCIMAPNoopOperation.h>
```
Comment (dinhviethoa) [inline src/async/imap/MCAsyncIMAP.h]: Why do you think it should be public?

**Comment by dinhviethoa** [comment]
The build failure is due to the cmake build workflow.

**Comment by conradev** [comment]
Hi Hoà!

The issue is that these headers are included in the built framework's `Headers` directory, but they are not referenced by the umbrella header. This causes a warning in Xcode. Perhaps the correct solution would be to make it so that these files don't end up in the `Headers` directory? I can make that change.

**Comment by dinhviethoa** [comment]
Thanks!

**Pull request merged**